[2025-07-18T22:02:19.724+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-07-18T22:02:19.732+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: python_pipeline.groupby_region manual__2025-07-18T22:02:07.120197+00:00 [queued]>
[2025-07-18T22:02:19.736+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: python_pipeline.groupby_region manual__2025-07-18T22:02:07.120197+00:00 [queued]>
[2025-07-18T22:02:19.736+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2025-07-18T22:02:19.742+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): groupby_region> on 2025-07-18 22:02:07.120197+00:00
[2025-07-18T22:02:19.748+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'python_pipeline', 'groupby_region', 'manual__2025-07-18T22:02:07.120197+00:00', '--job-id', '8', '--raw', '--subdir', 'DAGS_FOLDER/executing_python_pipeline.py', '--cfg-path', '/tmp/tmpo52ejny7']
[2025-07-18T22:02:19.750+0000] {standard_task_runner.py:105} INFO - Job 8: Subtask groupby_region
[2025-07-18T22:02:19.750+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=259) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-07-18T22:02:19.751+0000] {standard_task_runner.py:72} INFO - Started process 284 to run task
[2025-07-18T22:02:19.786+0000] {task_command.py:467} INFO - Running <TaskInstance: python_pipeline.groupby_region manual__2025-07-18T22:02:07.120197+00:00 [running]> on host 4e252374bd7b
[2025-07-18T22:02:19.828+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='python_pipeline' AIRFLOW_CTX_TASK_ID='groupby_region' AIRFLOW_CTX_EXECUTION_DATE='2025-07-18T22:02:07.120197+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-07-18T22:02:07.120197+00:00'
[2025-07-18T22:02:19.829+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-07-18T22:02:19.846+0000] {logging_mixin.py:190} WARNING - /opt/***/dags/executing_python_pipeline.py:40 FutureWarning: Passing literal json to 'read_json' is deprecated and will be removed in a future version. To read from a literal string, wrap it in a 'StringIO' object.
[2025-07-18T22:02:19.858+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/executing_python_pipeline.py", line 46, in groupby_region
    region_df.to_csv(
    ^^^^^^^^^^^^^^^^
AttributeError: 'function' object has no attribute 'to_csv'
[2025-07-18T22:02:19.875+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=python_pipeline, task_id=groupby_region, run_id=manual__2025-07-18T22:02:07.120197+00:00, execution_date=20250718T220207, start_date=20250718T220219, end_date=20250718T220219
[2025-07-18T22:02:19.882+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-07-18T22:02:19.882+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 8 for task groupby_region ('function' object has no attribute 'to_csv'; 284)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/executing_python_pipeline.py", line 46, in groupby_region
    region_df.to_csv(
    ^^^^^^^^^^^^^^^^
AttributeError: 'function' object has no attribute 'to_csv'
[2025-07-18T22:02:19.924+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-07-18T22:02:19.934+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-07-18T22:02:19.935+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
